### PR TITLE
Ensure that new functions update in the dev server

### PR DESCRIPTION
**Pull request by tonyhb**
This uses deterministic app IDs + function IDs, then recreates the list of functions and apps when registering.

Deterministic IDs ensure that things are persistent across mutations (eg. function run history).

## Type of change (choose one)
- [ ] Chore (refactors, upgrades, etc.)
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: pkg/devserver/api.go [hunk 1]
```diff
@@ -161,7 +161,8 @@ func (a devapi) register(ctx context.Context, r sdk.RegisterRequest) (err error)
 
 	// We need a UUID to register functions with.
 	appParams := cqrs.InsertAppParams{
-		ID:          uuid.New(),
+		// Use a deterministic ID for the app in dev.
+		ID:          uuid.NewSHA1(uuid.NameSpaceOID, []byte(r.URL)),
```
Comment (djfarrelly) [inline pkg/devserver/api.go:165]: This is much better!

File: pkg/cqrs/functions.go [hunk 2]
```diff
@@ -40,6 +40,7 @@ type FunctionReader interface {
 
 type FunctionWriter interface {
 	InsertFunction(ctx context.Context, params InsertFunctionParams) (*Function, error)
+	DeleteFunctionsByAppID(ctx context.Context, appID uuid.UUID) error
```
Comment (darwin67) [inline pkg/cqrs/functions.go:43]: :tada: 
love it

File: pkg/devserver/api.go [hunk 3]
```diff
@@ -173,6 +174,13 @@ func (a devapi) register(ctx context.Context, r sdk.RegisterRequest) (err error)
 		Checksum: sum,
 	}
 
+	// Initially, we must delete all functions because we're straight up replacing them.
+	// This allows us to clean functions that are removed. Functions have a deterministic ID
+	// and so logs etc. are all still persisted.
+	if err := a.devserver.data.DeleteFunctionsByAppID(ctx, appParams.ID); err != nil {
```
Comment (darwin67) [inline pkg/devserver/api.go:180]: shouldn't this be inside the transaction as well?

**Review by djfarrelly — APPROVED** [review]

File: pkg/devserver/api.go (hunk 3 above)
Comment (tonyhb) [reply pkg/devserver/api.go]: DuckDB doesn't handle txns with primary keys, but I've fixed this by loosening constraints.

Comment (darwin67) [reply pkg/devserver/api.go]: this is not the kind of data that needs to be in a columnal DB. I wonder if cqrs should have separate interfaces for relational DB vs columnal DB.

**Review by darwin67 — APPROVED** [review]

**Pull request merged**